pkg/controller/servicemesh/webhooks/mutation/controlplane.go

```diff
@@ -60,34 +60,45 @@ func (v *ControlPlaneMutator) Handle(ctx context.Context, req admission.Request)
 	// on create we set the version to the current default version
 	// on update, if the version is removed we reset it to what was previously set
 	currentVersion := mutator.NewVersion()
-	effectiveVersion, _ := versions.ParseVersion(currentVersion)
 	if currentVersion == "" {
 		switch req.AdmissionRequest.Operation {
 		case admissionv1beta1.Create:
 			log.Info("Setting .spec.version to default value", "version", versions.DefaultVersion.String())
-			mutator.SetVersion(mutator.DefaultVersion())
-			effectiveVersion, _ = versions.ParseVersion(mutator.DefaultVersion())
+			currentVersion = mutator.DefaultVersion()
 		case admissionv1beta1.Update:
 			oldVersion := mutator.OldVersion()
 			if currentVersion != oldVersion && oldVersion != versions.InvalidVersion.String() {
 				log.Info("Setting .spec.version to existing value", "version", oldVersion)
-				mutator.SetVersion(oldVersion)
-				effectiveVersion, _ = versions.ParseVersion(oldVersion)
+				currentVersion = oldVersion
 			}
 		}
 	}
+	mutator.SetVersion(currentVersion)
```

Comment:
Does this work when the version is already set, given the fact that the patch generated by `SetVersion` adds the `version` field? Previously, we only called this function when the `version` field wasn't set.